diskmaker/controllers/deleter/reconcile.go

```diff
@@ -87,6 +88,11 @@ func (r *DeleteReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return ctrl.Result{}, fmt.Errorf("failed to initialize PV cache: %w", err)
 		}
 		for _, pv := range pvList.Items {
+			err = clearUIDFromPV(&pv, r.Client, r.runtimeConfig)
```

Comment:
What if `node.Name` looks like an UID? You will delete it here. IMO there should be a generic function `pvMatchesNode(pv, node)`, that would check the annotation is either `local-volume-provisioner-NODE_NAME` **or** its starts with `local-volume-provisioner-NODE_NAME` and the rest looks like an UUID.

And I'd use the fuction everywhere where the annotation is evaluated and current node matched, not only in the deleter. I would not save the PV in the API server, but it's matter of style, it's not bad to save it. Still, diskmaker should only update PVs it owns, not PVs on the other nodes.

Reply:
Thanks for the review, posting new version.